externs/es3.js

```diff
@@ -141,6 +141,18 @@ function IArrayLike() {}
 /** @type {number} */
 IArrayLike.prototype.length;
 
+/**
+ * @record
+ * @extends {IArrayLike.<string>}
```

Comment:
Using IArrayLike is actually a problem here, because the result is really an Array and client code should be able to call array methods on it.

I think the only way to do this will be:

```javascript
/**
 * @constructor
 * @extends {Array<string>}
 */
function IRegExpResult() {};
```

This makes it look like you can call `new IRegExpResult()` when you really can't, but I think that's OK in this case.